File: src/main/java/org/kite9/diagram/visualization/planarization/transform/ExcessVertexRemovalTransform.java
package org.kite9.diagram.visualization.planarization.transform;

import java.util.ArrayList;
import java.util.List;

import org.kite9.diagram.common.elements.Vertex;
import org.kite9.diagram.visualization.planarization.Planarization;
import org.kite9.diagram.visualization.planarization.Tools;
import org.kite9.framework.logging.Logable;


/**
 * This performs the following transformations: 
 * <ol>
 * <li> Remove excess start/end temporary vertices</li>
 * <li> Remove any dimensionless vertex with only two edges incident to it
 * </ol>
 * 
 * 
 * @author robmoffat
 *
 */
public class ExcessVertexRemovalTransform implements PlanarizationTransform, Logable {

	public void transform(Planarization pln) {
		removeExcessVertices(pln);

	}
	
	private void removeExcessVertices(Planarization pln) {
		Tools t = new Tools();
		List<Object> vertices = new ArrayList<Object>(pln.getEdgeOrderings().keySet());
		for (Object v : vertices) {
			if (v instanceof Vertex) {
				t.checkRemoveVertex(pln, (Vertex) v);
			}
		}
	}

	

	public String getPrefix() {
		return "EVRT";
	}

	public boolean isLoggingEnabled() {
		return false;
	}

}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/transform/ExcessVertexRemovalTransform.kt
package org.kite9.diagram.visualization.planarization.transform

import org.kite9.diagram.common.elements.vertex.Vertex
import org.kite9.diagram.logging.Logable
import org.kite9.diagram.visualization.planarization.Planarization
import org.kite9.diagram.visualization.planarization.Tools

/**
 * This performs the following transformations:
 *
 *  1.  Remove excess start/end temporary vertices
 *  1.  Remove any dimensionless vertex with only two edges incident to it
 *
 *
 *
 * @author robmoffat
 */
class ExcessVertexRemovalTransform : PlanarizationTransform, Logable {

    override fun transform(pln: Planarization) {
        removeExcessVertices(pln)
    }

    private fun removeExcessVertices(pln: Planarization) {
        val t = Tools()
        val vertices: List<Any> = ArrayList<Any>(pln.edgeOrderings.keys)
        for (v in vertices) {
            if (v is Vertex) {
                t.checkRemoveVertex(pln, v)
            }
        }
    }

    override val prefix: String
        get() = "EVRT"
    override val isLoggingEnabled: Boolean
        get() = false
}